src/libraries/System.Private.Xml/src/System/Xml/Xsl/Xslt/XsltLoader.cs

```diff
@@ -59,7 +59,7 @@ public void Load(Compiler compiler, object stylesheet, XmlResolver? xmlResolver)
                 {
                     // If xmlResolver == null, then the original uri will be resolved using XmlUrlResolver
                     XmlResolver origResolver = xmlResolver!;
-                    if (xmlResolver == null || xmlResolver == XmlNullResolver.Singleton)
+                    if (xmlResolver == null || xmlResolver == XmlResolver.ThrowingResolver)
                         origResolver = new XmlUrlResolver();
```

Comment:
this seems a bit weird, if user explicitly chooses throwing resolver we will still go with XmlUrlResolver?

Reply:
Yeah, thought the same. It looks like a bug, but perhaps I was being too deferential to preserving back-compat. Maybe killing that clause is the best option.

Reply:
yes, I'd honestly go with just:
`xmlResolver ??= XmlResolver.ThrowingResolver` or `xmlResolver ??= new XmlUrlResolver()` depending if we expect unsafe inputs in those API

Reply:
Looks like that line is meant to override whatever the "default" resolver is. For example, if you call `XslCompiledTransform.Load(string)`, it will eventually hit:

https://github.com/dotnet/runtime/blob/e71a9583b4d6c9bd97edd87cda7f98f232f63530/src/libraries/System.Private.Xml/src/System/Xml/Xslt/XslTransform.cs#L259-L269

And the "ignore the null resolver" logic was apparently meant to work around this. This seems completely broken to me, since why bother putting in a secure default if we're just going to ignore it?

Reply:
Looks like it was added because the URL to the stylesheet itself (the arg passed to `XslCompiledTransform.Load(string)`) also goes through the resolver for some reason. The system needs to strip it off temporarily - just while loading the outermost stylesheet - so that resolution can take place.